src/cpp/src/visual_language/pipeline.cpp

```diff
@@ -183,6 +189,14 @@ class VLMPipeline::VLMPipelineImpl : public VLMPipelineBase{
         ov::Tensor inputs_embeds = m_inputs_embedder->get_inputs_embeds(prompt, rgbs, perf_metrics);
         auto end_get_inputs_embeds = std::chrono::steady_clock::now();
 
+        if (m_is_npu) {
+            // Prefill model in NPU is reshaped to NPUW_LLM_MAX_PROMPT_LEN x NPUW_LLM_MAX_PROMPT_LEN
+            OPENVINO_ASSERT(inputs_embeds.get_shape().at(1) <= m_max_prompt_len,
+                "VLM pipeline on NPU may only process input embeddings up to ", m_max_prompt_len,
+                " tokens. ", inputs_embeds.get_shape().at(1), " is passed.\nSet the \"MAX_PROMPT_LEN\""
+                " config option to increase the limit.");
```

Comment:
do we always expect to have the prompt size at 1st dim?

Reply:
I think yes, I've got this code from here: https://github.com/openvinotoolkit/openvino.genai/pull/1996/files/21f5402daafc019815416ad700f65b0ca4206f19#diff-3652c10cef326776cbe4b8b114c3fbc5b5dc90b0998dc7dd6ba5b1664e3d6194R209

Reply:
order of dimensions can be different for KV cache only.
typical inputs have an unified format.